components/http-client/src/api_client.rs

```diff
@@ -1,82 +1,96 @@
-use std::{path::Path,
+use std::{fs::{self,
+               File},
+          io::{Read,
+               Write},
+          iter::FromIterator,
+          path::{Path,
+                 PathBuf},
+          str::FromStr,
           time::Duration};
 
```

Comment:
Here's another newline we can remove, allowing `rustfmt` to keep everything sorted for us.